### PR TITLE
refactor(core): use bip32 in ecdh operations

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
* Move from class method to regular function (we don't use the bitgo
  object)
* Move tests
* Add support for bip32 arguments
* Return buffer, convert to `hex` at callsite

Issue: BG-34381

### CODE REVIEW

File: modules/core/src/bitgo.ts [hunk 1]
```diff
@@ -942,29 +937,6 @@ export class BitGo {
     return true;
   }
 
-  /**
-   * Construct an ECDH secret from a private key and other user's public key
-   */
-  getECDHSecret({ otherPubKeyHex, eckey }: GetEcdhSecretOptions): string {
```
Comment (tylerlevine) [inline modules/core/src/bitgo.ts:948]: Could you please keep this function on the `bitgo` object as a pass through to the new one in `ecdh.ts` for the moment (and mark as `@deprecated`)? There are at least two places in the UI where this is called directly.

We can remove in a future major version.

**Review by tylerlevine — CHANGES_REQUESTED** [review]

File: modules/core/src/bitgo.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/core/src/bitgo.ts:948]: ouch, forgot to check there

File: modules/core/src/ecdh.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  } else if (privateKey instanceof utxolib.ECPair) {
+    privateKey = privateKey.d.toBuffer(32);
+  }
+
+  if (!Buffer.isBuffer(publicKey)) {
+    publicKey = publicKey.publicKey;
+  }
+
+  if (!Buffer.isBuffer(privateKey) || !Buffer.isBuffer(publicKey)) {
+    throw new Error(`invalid state`);
```
Comment (sangaman) [inline modules/core/src/ecdh.ts:44]: Maybe add some more detail to this message?

**Review by sangaman — COMMENTED** [review]
Changes look good overall.

File: modules/core/src/ecdh.ts (hunk 2 above)
Comment (OttoAllmendinger) [reply modules/core/src/ecdh.ts:44]: This is more to satisfy `tsc` than anything else. You'd really have put in some effort to get here. I would have used `assert` but I didn't want the extra import.

Comment (OttoAllmendinger) [reply modules/core/src/ecdh.ts:44]: Wait a second, I actually do have `assert` already imported. I think for some reason the type assertion signature was not recognized. In any case, the user will almost certainly never see this and us devs have the stack trace.

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**